### ISSUE TITLE
[MU4 Issue] Improve tie direction in chords containing seconds

### ISSUE THREAD

**Issue by oktophonie** [musescore/MuseScore#8973]
As far as I can deduce, MuseScore has two primary rules for determining the direction of ties in chords (when 'Tie direction' = 'Auto'):

1. the uppermost tie curves up, the bottommost curves down, and those in the middle go up or down following the usual rules (if the note is on the middle stave line, or higher, the tie curves up, otherwise it curves down).
2. when the interval of a second is encountered in a chord, the tie to the bottom note always curves down, and the tie to the top note curves up

Ideally, the tie direction should flip only once in the chord; rule 2 causes a problem where the interval of a second appears more than once in the chord, since this means the direction changes more than once.

I recommend the following solution:

- adjust rule 1 to say: for a chord with an odd number of notes, the lower half curve down, the upper half curve up. If there is an odd number, the middle tie curves away from the stem (i.e. curving up if the tie is down, and vice versa); this happens to be the rule recommended by Gould(!)
- UNLESS the chord contains _one_ (and only one) instance of the interval of a second (or a unison), in which case the tie direction changes at that point (the tie to the upper note of the second/unison curves up, as do the ties of any higher notes; the tie to the lower note of the second/unison curves down, as do the ties of any lower notes).

Thus:

![chord-ties-1](https://user-images.githubusercontent.com/20604032/131549402-f886dbd2-ce46-40b9-9276-76d2bc97e6c7.png)

At first this does not appear to be a graphical improvement as there are many collisions with noteheads and between adjacent ties. This is, however, already a problem in many situations and will receive proper treatment in a separate issue (#8980).

(Naturally, this is all moot where there is more than one voice on the stave.)
